src/components/StageBar.vue

```diff
@@ -1,14 +1,14 @@
 <template>
   <div class="stage-bar">
-    <button
-      v-for="stage in stages"
+    <div
+      v-for="(stage, index) in stages"
       :key="stage.name"
-      class="link-btn"
-      :class="[current === stage.name ? 'active' : '']"
       @click="() => changeStage(stage.name)"
+      class="link-btn"
+      :class="[current === stage.name ? `active item-${index}` : `item-${index}`]"
```

Comment:
We can make the most here of Vue's nifty support for conditional classes:
```suggestion
      :class="{ active: current === stage.name }"
```